src/App.jsx

```diff
@@ -1,248 +1,364 @@
+                      type="button"
+                      className="delete"
+                      onClick={() => setQuery('')}
+                    />
+                  )
+                  }
                 </span>
-              </th>
-            </tr>
-          </thead>
-
-          <tbody>
-            <tr data-cy="Product">
-              <td className="has-text-weight-bold" data-cy="ProductId">
-                1
-              </td>
-
-              <td data-cy="ProductName">Milk</td>
-              <td data-cy="ProductCategory">🍺 - Drinks</td>
-
-              <td
-                data-cy="ProductUser"
-                className="has-text-link"
+              </p>
+            </div>
+
+            <div className="panel-block is-flex-wrap-wrap">
+
+              <a
+                href="#/"
+                data-cy="AllCategories"
+                className={cn('button mr-6 is-success',
+                  { 'is-outlined':
+                    filterByCategory.length !== 0 })}
+                onClick={() => setFilterByCategory([])}
               >
-                Max
-              </td>
-            </tr>
-
-            <tr data-cy="Product">
-              <td className="has-text-weight-bold" data-cy="ProductId">
-                2
-              </td>
-
-              <td data-cy="ProductName">Bread</td>
-              <td data-cy="ProductCategory">🍞 - Grocery</td>
-
-              <td
-                data-cy="ProductUser"
-                className="has-text-danger"
+                All
+              </a>
+
+              {categoriesFromServer.map((category) => {
+                const isSelected = filterByCategory.includes(category.title);
+
+                return (
+                  <a
+                    data-cy="Category"
+                    className={cn('button mr-2 my-1',
+                      { 'is-info': filterByCategory.includes(category.title) })}
+                    href="#/"
+                    key={category.id}
+                    onClick={() => {
+                      if (isSelected) {
+                        setFilterByCategory(filterByCategory.filter(
+                          categor => categor.title !== category.title,
+                        ));
+                      }
+
+                      setFilterByCategory(
+                        [...filterByCategory, category.title],
+                      );
+                    }
+                }
+                  >
+                    {category.title}
+                  </a>
+                );
+              })}
+            </div>
+
+            <div className="panel-block">
+              <a
+                data-cy="ResetAllButton"
+                href="#/"
+                className="button is-link is-outlined is-fullwidth"
+                onClick={() => {
+                  setFilterByName(FILTER_BY.NAME.ALL);
+                  setFilterByCategory([]);
+                  setQuery('');
+                }}
```

Comment:
If function takes more than 1 line you should declare it in component body and pass here